modules/home-environment.nix

```
@@ -592,10 +592,21 @@ in
         ''
       else
         ''
+          function hm-internal-replace-profile {
+
```

Function names should use camel casing, also you have an accidental empty line.

```suggestion
          function nixReplaceProfile() {
```

Also, do `unset -f nixReplaceProfile` at the end of the activation block.